common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -161,6 +162,182 @@ public void saveTokens(
         saveCredentials(accessTokenToSave, refreshTokenToSave, idTokenToSave);
     }
 
+    @Override
+    public Account getAccount(@NonNull final String environment,
+                              @NonNull final String clientId,
+                              @NonNull final String homeAccountId) {
+        final List<Account> allAccounts = getAccounts(environment, clientId);
+
+        // Return the sought Account matching the supplied homeAccountId
+        for (final Account account : allAccounts) {
+            if (homeAccountId.equals(account.getHomeAccountId())) {
+                return account;
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public List<Account> getAccounts(@NonNull final String environment,
+                                     @NonNull final String clientId) {
+        final List<Account> accountsForThisApp = new ArrayList<>();
+
+        // Get all of the Accounts for this environment
+        final List<Account> accountsForEnvironment =
+                mAccountCredentialCache.getAccountsFilteredBy(
+                        null, // wildcard (*) homeAccountId
+                        environment,
+                        null // wildcard (*) realm
+                );
+
+        // Declare a List to hold the MicrosoftStsAccounts
+        final List<Account> microsoftStsAccounts = new ArrayList<>();
```

Comment:
We probably shouldn't assume these accounts are MSTS accounts.  What about ADFS, B2C?

Reply:
Resolved via ea7e9ee4b7eb23135ad0223021b7dcfa5175e817